packages/zipkin-instrumentation-restify/src/restifyMiddleware.js

```diff
@@ -82,13 +82,19 @@ module.exports = function restifyMiddleware({tracer, serviceName = 'unknown', po
         tracer.recordBinary(Header.Flags, id.flags.toString());
       }
 
-      res.on('finish', () => {
+      const onCloseOrFinish = () => {
+        res.removeListener('close', onCloseOrFinish);
```

Comment:
Yes but the issue is that either `close` or `finish` get triggered so I opted remove both just to be safe.